src/MemPool.jl

```diff
@@ -100,39 +100,99 @@ function approx_size(xs::AbstractArray{String})
     s + 4 * length(xs)
 end
 
-function approx_size(s::String) 
+function approx_size(s::String)
     sizeof(s)+sizeof(Int) # sizeof(Int) for 64 bit vs 32 bit systems
 end
 
-function approx_size(s::Symbol) 
+function approx_size(s::Symbol)
     sizeof(s)+sizeof(Int)
 end
 
+struct DiskCacheConfig
+    toggle::Bool
+    membound::Int
+    diskpath::AbstractString
+    diskdevice::StorageDevice
+    diskbound::Int
+    allocator_type::AbstractString
+    evict_delay::Int
+end
+
+function DiskCacheConfig(;
+    toggle::Bool=false,
+    membound::Union{Nothing,Int}=nothing,
+    diskpath::Union{Nothing,AbstractString}=nothing,
+    diskdevice::Union{Nothing,StorageDevice}=nothing,
+    diskbound::Union{Nothing,Int}=nothing,
+    allocator_type::Union{Nothing,AbstractString}=nothing,
+    evict_delay::Union{Nothing,Int}=nothing,
+)
+    toggle = something(toggle, get(ENV, "JULIA_MEMPOOL_EXPERIMENTAL_FANCY_ALLOCATOR", "0"))
```

Comment:
This will always take the value of `toggle` because it's required to be a `Bool` above.

Reply:
fixed